### PR TITLE
Patches for Sagemaker v2

### PR DESCRIPTION
**Pull request by jagpk**

### CODE REVIEW

**Comment by jagpk** [comment]
Updated Notebook, train.py. inference.py and config.py to work with latest Sagemaker version.

File: question_answering/src/config.py [hunk 1]
```diff
@@ -14,7 +14,7 @@
 
 MODEL_CONFIG_CLASSES = list(txf.MODEL_FOR_QUESTION_ANSWERING_MAPPING.keys())
 MODEL_TYPES = tuple(conf.model_type for conf in MODEL_CONFIG_CLASSES)
-ALL_MODELS = sum((tuple(conf.pretrained_config_archive_map.keys()) for conf in MODEL_CONFIG_CLASSES), (),)
+#ALL_MODELS = sum((tuple(conf.pretrained_config_archive_map.keys()) for conf in MODEL_CONFIG_CLASSES), (),)
```
Comment (athewsey) [inline question_answering/src/config.py]: Can just delete this instead of commenting out?

File: question_answering/src/config.py [hunk 2]
```diff
@@ -57,7 +57,7 @@ def parse_args():
     parser.add_argument(
         "--config-name", type=str,
         default=hps.get("config-name", "bert-base-uncased"),
-        help=f"Path to pre-trained model or shortcut name selected in the list: {', '.join(ALL_MODELS)}"
+        #help=f"Path to pre-trained model or shortcut name selected in the list: {', '.join(ALL_MODELS)}"
```
Comment (athewsey) [inline question_answering/src/config.py]: Please keep the help message in, just cut the part where we list out the models? E.g. just something like "Path to pre-trained model or HuggingFace model config name" would be fine I think?

File: question_answering/src/inference.py [hunk 3]
```diff
@@ -33,5 +33,4 @@ def predict_fn(input_data, model):
     trained_model, tokenizer, config = model
     nlp = pipeline(task="question-answering", model=trained_model, config=config, tokenizer=tokenizer, framework="pt")
     result = nlp(input_data)
-    return result
-    
+    return result
```
Comment (athewsey) [inline question_answering/src/inference.py]: Scripts should end with empty line (no trailing whitespace, just newline... And sorry we don't have an auto-linter in this repo)

**Review by athewsey — COMMENTED** [review]
Thanks! A few small comments it'd be nice to address if you can? Otherwise I can just merge and fix forward

**Review by athewsey — APPROVED** [review]
Great, thank you!

**Pull request merged**